frontend/components/portfolio/group/membership/MenuButton.vue

```diff
@@ -0,0 +1,175 @@
+<script setup lang="ts">
+import type OverlayPanel from 'primevue/overlaypanel'
+import { useToast } from 'primevue/usetoast'
+import { type Portfolio, type PortfolioGroup } from '@/openapi/generated/pacta'
+
+const { t } = useI18n()
+const toast = useToast()
+const { newPortfolioGroup: { newPortfolioGroupVisible } } = useModal()
+const pactaClient = usePACTA()
+
+interface Props {
+  portfolioGroups: PortfolioGroup[]
+  selectedPortfolios: Portfolio[]
+}
+const props = defineProps<Props>()
+interface Emits {
+  (e: 'changed-memberships'): void
+  (e: 'changed-groups'): void
+}
+const emit = defineEmits<Emits>()
+
+const prefix = 'components/portfolio/group/membership/MenuButton'
+const tt = (s: string) => t(`${prefix}.${s}`)
+const visible = useState<boolean>(`${prefix}.visible`, () => false)
+
+const overlayPanel = useState<OverlayPanel>(`${prefix}.overlayPanel`)
+const toggleMenu = (event: Event) => {
+  presentOrFileBug(overlayPanel.value).toggle(event)
+  visible.value = !visible.value
+}
+const hideMenu = () => {
+  visible.value = false
+}
+const showMenu = () => {
+  visible.value = true
+}
+
+const changeMemberships = (portfolioGroupId: string, add: boolean) => {
+  return async (event: Event) => {
+    const portfolioIds = props.selectedPortfolios.map((portfolio) => portfolio.id)
+    if (add) {
+      await Promise.all(portfolioIds.map((portfolioId) => pactaClient.createPortfolioGroupMembership({ portfolioId, portfolioGroupId })))
+    } else {
+      await Promise.all(portfolioIds.map((portfolioId) => pactaClient.deletePortfolioGroupMembership({ portfolioId, portfolioGroupId })))
+    }
+    emit('changed-memberships')
+    toast.add({
+      severity: 'success',
+      group: add ? 'portfolio-group-membership-added' : 'portfolio-group-membership-removed',
+    })
+    // toggleMenu(event)
+  }
+}
+const changedGroups = () => {
+  emit('changed-groups')
+}
+
+type Icon = 'empty' | 'partial' | 'full'
+
+const groupOptions = computed(() => {
+  const selected = props.selectedPortfolios
+  const result = props.portfolioGroups.map((pg) => {
+    const isMember = selected.map((portfolio) => (portfolio.groups ?? []).some((pg2) => pg.id === pg2.portfolioGroup.id))
+    const anySelected = isMember.some(m => m)
+    const allSelected = isMember.every(m => m)
+    let icon: Icon = 'empty'
+    let addIfClick = true
+    let hoverText = tt('Add all portfolios to group')
+    if (allSelected) {
+      icon = 'full'
+      addIfClick = false
+      hoverText = tt('Remove all portfolios from group')
+    } else if (anySelected) {
+      // TODO(grady) make a pi-square-minus
+      icon = 'partial'
+      addIfClick = true
+      hoverText = tt('Add unselected portfolios to group')
+    }
+    return {
+      id: pg.id,
+      label: pg.name,
+      icon,
+      cmd: changeMemberships(pg.id, addIfClick),
+      hoverText,
+      created: pg.createdAt,
+    }
+  })
+  result.sort((a, b) => a.created < b.created ? 1 : -1)
```

Comment:
Not sure what the type of `created` is, but one trick for sorting things is to do `a.created - b.created` or vice versa. I think the cardinality helps it sort more efficiently too.

Reply:
It's an ISO string, so it's lexographically accurate, which is great, and we don't need to parse it to do the comparison. I'll comment something to that effect.